technic/register.lua

```diff
@@ -7,7 +7,7 @@ technic.producer_receiver = "PR_RE"
 technic.battery  = "BA"
 
 technic.machines    = {}
-technic.power_tools = {}
+technic.power_tools = {} -- TODO: Should get rid of this table, tool stack already has all required data and more
```

Comment:
Probably better to keep it, if it's removed then something like this will have to be done wherever it was used:
```diff
- local max_charge = technic.power_tools[stack:get_name()]
+ local max_charge = technic.power_tools and technic.power_tools[stack:get_name()]
+     or stack:get_definition().max_charge or 10000
```

Reply:
> Probably better to keep it, if it's removed then something like this will have to be done wherever it was used:
> 
> ```diff
> - local max_charge = technic.power_tools[stack:get_name()]
> + local max_charge = technic.power_tools and technic.power_tools[stack:get_name()]
> +     or stack:get_definition().max_charge or 10000
> ```

It could be removed without doing that, reason why I left it and only marked with comment was because other mods might use it and there's already enough breaking changes.
This table however is useless for Technic Plus, only reason to keep it is some small backwards compatibility with official technic.

Reply:
Should probably also say here that `stack:get_definition().max_charge or 10000` used in registration and compatibility shims is already bit paranoid way to avoid errors...
And actually `or 10000` should not be needed ever, there simply should not be path to allow this unless functions are called with stack that is not registered as power tool. In that case previous implementation would also have failed in few cases and really problem is in mod calling these functions if stack is not valid tool.

Actually I've been thinking all the time since I added those about removing `or 10000` and instead let it crash.